rllib/algorithms/impala/torch/impala_torch_learner.py

```diff
@@ -0,0 +1,121 @@
+from typing import Mapping
+
+import numpy as np
+
+from ray.rllib.algorithms.impala.impala_base_learner import ImpalaBaseLearner
+from ray.rllib.algorithms.impala.torch.vtrace_torch_v2 import (
+    vtrace_torch,
+    make_time_major,
+)
+from ray.rllib.core.learner.torch.torch_learner import TorchLearner
+from ray.rllib.policy.sample_batch import SampleBatch
+from ray.rllib.utils.annotations import override
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.torch_utils import convert_to_torch_tensor
+from ray.rllib.utils.typing import TensorType
+
+torch, nn = try_import_torch()
+
+
+class ImpalaTorchLearner(TorchLearner, ImpalaBaseLearner):
+    """Implements the IMPALA loss function in torch."""
+
+    def __init__(self, *args, **kwargs):
+        TorchLearner.__init__(self, *args, **kwargs)
+        ImpalaBaseLearner.__init__(self, *args, **kwargs)
+
+    @override(TorchLearner)
+    def compute_loss_per_module(
+        self, module_id: str, batch: SampleBatch, fwd_out: Mapping[str, TensorType]
+    ) -> TensorType:
+        target_policy_dist = fwd_out[SampleBatch.ACTION_DIST]
+        values = fwd_out[SampleBatch.VF_PREDS]
+
+        # TODO(Artur): Why are we missing batch[SampleBatch.ACTION_LOGP] here?
```

Comment:
What does this question imply? is this note still relevant?

Reply:
Sorry, this TODO got resolved.